lib/active_merchant/billing/gateways/worldpay.rb

```diff
@@ -241,7 +268,7 @@ def add_amount(xml, money, options)
       end
 
       def add_payment_method(xml, amount, payment_method, options)
-        if payment_method.is_a?(String)
+        if options[:payment_type] == :pay_as_order
```

Comment:
I have yet to fully read your additions, but I'm seeing that for AM users of this feature, they'll need to start sending a new `payment_type` option? Would it hurt to retain the string check for backward compatibility?

Reply:
Or was this made to distinguish it from stored PMs, since they end up as strings?

Reply:
I see now that this is a flag created by a method that parses the PM, neat. Disregard for now.

Reply:
Yep, you got it. The problem was that both the TPV tokens and pay-by-reference/pay-as-order string payment methods needed to be distinguished early on. Not a user-facing option, should be fully backwards-compatible.